# AirBnb Price Prediction
- Understanding problem statement
- Data Collection
- Data Checks
- Exploratory Data Analysis
- Data Pre-processing
- Model Training
- Choose the best model

## 1. Problem Statement

- The project understands how the price of AirBnb properties is affected with variables such location, property type , Bed type, Availability

## 2. Data Collection

- Data Source - https://www.kaggle.com/datasets/joebeachcapital/airbnb
- Dataset consist of 89 columns and 75k records, we be only using relevent columns.

In [1]:
import pandas as pd

In [39]:
df = pd.read_csv('airbnb-listings.csv',delimiter=';',low_memory=False,usecols=['Price', 'Smart Location', 'Room Type', 'Property Type', 'Bed Type', 'Availability 365', 'Minimum Nights', 
                 'Number of Reviews', 'Review Scores Rating', 'Cancellation Policy'])

'A'

In [44]:
df.columns = [col.replace(" ","").lower() for col in df.columns]

In [45]:
df.head()

smartlocation propertytype         roomtype   bedtype  price  \
0  Amsterdam, Netherlands    Apartment  Entire home/apt  Real Bed  600.0   
1  Amsterdam, Netherlands    Apartment  Entire home/apt  Real Bed  175.0   
2  Amsterdam, Netherlands    Apartment  Entire home/apt  Real Bed  125.0   
3  Amsterdam, Netherlands    Apartment  Entire home/apt  Real Bed  130.0   
4  Amsterdam, Netherlands    Apartment  Entire home/apt  Real Bed   80.0   

   minimumnights  availability365  numberofreviews  reviewscoresrating  \
0            2.0             74.0             31.0                89.0   
1            2.0            259.0             15.0                99.0   
2            4.0              0.0              1.0               100.0   
3            3.0              0.0             22.0                97.0   
4            2.0            326.0             16.0                78.0   

  cancellationpolicy  
0             strict  
1             strict  
2           flexible  
3           flexible  
4           moderate

In [46]:
df.tail()

smartlocation propertytype         roomtype   bedtype  \
494949       Footscray, Australia        House     Private room  Real Bed   
494950      Yarraville, Australia        House     Private room  Real Bed   
494951  West Footscray, Australia        House     Private room  Real Bed   
494952       Footscray, Australia    Apartment  Entire home/apt  Real Bed   
494953      Kingsville, Australia        House     Private room  Real Bed   

        price  minimumnights  availability365  numberofreviews  \
494949   30.0            1.0              0.0              4.0   
494950   42.0            7.0            358.0              9.0   
494951   33.0            3.0            337.0              3.0   
494952   62.0            5.0             28.0              0.0   
494953   47.0            2.0             63.0              0.0   

        reviewscoresrating cancellationpolicy  
494949                60.0           moderate  
494950                93.0           moderate  
494951               100.0             strict  
494952                 NaN           flexible  
494953                 NaN           moderate

### 2.2 Data Information
- smartlocation = location of the property i.e. city and country
- propertytype - type of property (house or apartment)
- roomtype - type of room
- bedtype - type of bed
- price - the target column
- minimumnights - minimum nights stays
- availability365 - availablity across year
- numberofreviews - total number of reviews recieved
- reviewscoresrating - rating of reviews
- cancellationpolicy - how rigid are the cancelation policies



In [47]:
## shape of dataset
df.shape

(494954, 10)

### 3. Data Checks
- Missing Values
- Duplicates
- Datatypes of variables
- check number of unique values for each column
- Check Statistics of Dataset
- check various categories present in the categorical columns

### 3.1 Missing Values



In [48]:
df.isnull().mean()

smartlocation         0.000004
propertytype          0.000022
roomtype              0.000002
bedtype               0.000002
price                 0.016078
minimumnights         0.000004
availability365       0.000004
numberofreviews       0.000004
reviewscoresrating    0.258246
cancellationpolicy    0.000004
dtype: float64

#### Observation
- We can see most of the columns have some missing values, we will all missing values

In [50]:
print(f"shape with missing values {df.shape}\n")

df.dropna(inplace=True)

print(f"shape after removing missing values {df.shape}")

shape with missing values (494954, 10)

shape after removing missing values (362420, 10)


### 3.2 Check Duplicates



In [51]:
df.duplicated().sum()

4302

#### Observation
- there are 4302 duplicates, we need to drop

In [53]:
print(f"shape with duplictes {df.shape}\n")

df.drop_duplicates(inplace=True)

print(f"shape after removing duplicates {df.shape}")

shape with duplictes (362420, 10)

shape after removing duplicates (358118, 10)


### 3.3 Check Data types

In [54]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 358118 entries, 0 to 494951
Data columns (total 10 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   smartlocation       358118 non-null  object 
 1   propertytype        358118 non-null  object 
 2   roomtype            358118 non-null  object 
 3   bedtype             358118 non-null  object 
 4   price               358118 non-null  float64
 5   minimumnights       358118 non-null  float64
 6   availability365     358118 non-null  float64
 7   numberofreviews     358118 non-null  float64
 8   reviewscoresrating  358118 non-null  float64
 9   cancellationpolicy  358118 non-null  object 
dtypes: float64(5), object(5)
memory usage: 30.1+ MB
